Dockerfile

```diff
@@ -0,0 +1,55 @@
+# One Dockerfile to rule them all
+FROM --platform=$BUILDPLATFORM golang:1.23.3 AS builder
+
+ARG APP
+ARG CGOENABLED
+ARG BUILDPLATFORM
+ARG TARGETPLATFORM
+ARG TARGETOS
+ARG TARGETARCH
+
+ENV GOOS=$TARGETOS \
+    GOARCH=$TARGETARCH \
+    GOEXPERIMENT=boringcrypto \
+    CGO_ENABLED=$CGOENABLED
+
+WORKDIR /workspace
+
+# Install cross-compilation tools
+RUN apt-get update && apt-get install -y \
+    gcc-x86-64-linux-gnu \
+    g++-x86-64-linux-gnu \
+    gcc-aarch64-linux-gnu \
+    g++-aarch64-linux-gnu \
+    libc6-dev-arm64-cross \
+    libc6-dev-amd64-cross
+
+
+# Download dependencies first (better layer caching)
+COPY go.mod go.sum ./
+RUN go mod download
+
+# Copy the app source code
+COPY . .
+
+
+# Build the app for the target architecture
+RUN echo "Building $APP on $BUILDPLATFORM targeting $TARGETPLATFORM"
+RUN if [ "$TARGETARCH" = "amd64" ]; then \
+        CC=x86_64-linux-gnu-gcc go build -o $APP /workspace/app/$APP/cmd/main.go; \
+    elif [ "$TARGETARCH" = "arm64" ]; then \
+        CC=aarch64-linux-gnu-gcc go build -o $APP /workspace/app/$APP/cmd/main.go; \
+    fi
+
+
+# Target distroless base image for CGO_ENABLED apps
+# This image includes a basic runtime environment with libc and other minimal dependencies
+FROM gcr.io/distroless/base AS distroless-base
+COPY --from=builder /workspace/$APP /binary
+ENTRYPOINT ["/binary"]
```

Comment:
better to name the binary as `nexus`, `spike`, or `keeper` instead of `binary`.

Because we'll likely shell into some of those, similar to this:

```bash
kubectl exec $podName -- spike secret list
```

There `binary secret list` would look a bit weird.

Reply:
we cannot use variables in ENTRYPOINT so when named to "nexus" for example we cant set the entrypoint to ["/nexus"] sinc we cant get $APP arg to work in entrypoint

Reply:
May make sense then to have 3 separate Dockerfiles.

Reply:
I am on the "separatre docker files" train too. -- that's how we ended up in VSecM too. -- each binary having its odnw dockerfile.

Reply:
put them all in a folder called kubernetes? where we would also have helms?

Reply:
Example: https://github.com/vmware-tanzu/secrets-manager/tree/main/dockerfiles/vsecm-ist-fips

It's a bit repetitive but also "[A little copying is better than a little dependency](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=9m28s))".

Reply:
i created an issu

https://github.com/spiffe/spike/issues/103